File: internal/search/testdata/sample.js
// Sample JavaScript file for testing

// Exported function
export function exportedFunction(param) {
  console.log(param);
  return param.toUpperCase();
}

// Non-exported function
function privateFunction() {
  return 'private';
}

// Exported class
export class ExportedClass {
  constructor(name) {
    this.name = name;
    this._privateProp = null;
  }

  publicMethod() {
    return this.name;
  }

  _privateMethod() {
    return this._privateProp;
  }
}

// Non-exported class
class PrivateClass {
  constructor() {
    this.value = 0;
  }
}

// Exported variable
export const EXPORTED_CONST = 'public';

// Non-exported variable
const PRIVATE_CONST = 'private';

let globalState = null;

File: internal/search/testdata/sample.py
# Sample Python file for testing

def public_function(param):
    """Public function with docstring"""
    print(param)
    return param.upper()

def _private_function():
    """Private function (by convention)"""
    return 'private'

class PublicClass:
    """Public class with docstring"""

    def __init__(self, name):
        self.name = name
        self._private_attr = None

    def public_method(self):
        """Public method"""
        return self.name

    def _private_method(self):
        """Private method (by convention)"""
        return self._private_attr

class _PrivateClass:
    """Private class (by convention)"""

    def __init__(self):
        self.value = 0

# Public constant
PUBLIC_CONST = 'public'

# Private constant (by convention)
_PRIVATE_CONST = 'private'

# Module-level variable
global_state = None
